### ISSUE TITLE
How to import all keys from specific path

### ISSUE THREAD

**Issue by Eslamanwar** [hashicorp/vault-action#234]
Is there a way to import all keys from specific path

**Comment by slemme1** [comment]
I was going to open another enhancement request to read all secrets in the provided path without specifying each one in the secrets input:
Example:
secrets:
/secrets/myPath * | prefix #get all secrets, using prefix+keyname as the output/env variable (or just keyname if no prefix).
Adding this feature would address the issue of setting secret names manually.
You can already gather all outputs and get them into json (and write to a file outside of the vault-action step):
env:
ALL_SECRETS_JSON: "${{ toJson(steps.vaultaction.outputs) }}"

I think this enhancement would just take changes in secrets.js getSecrets():
if {request.selector = *), remove this request and add a new one for each key in the path (populating request.envVarName, request.outputVarName as appropriate if prefix is supplied).
I have not contributed to open source before and would be glad to take this on if I can get some mentoring on the process...

**Comment by dcmoore-gd** [comment]
Is this anything like you're looking for?  https://github.com/hashicorp/vault-action/issues/153#issuecomment-824329289

**Comment by slemme1** [comment]
The issue with keys having dashes (e.g., "foo-bar") was reported in Nov 2020 and I think that was fixed because the automated tests (in v 2.3.0) have a key of "other-Secret-dash" that passes (so I am not sure why #153 was not closed). 

The implementation I just developed to read all keys in a path uses the same logic as individual keys and does not return a JSON output.  If the wildcard * is passed, the JSON returned for the path is parsed to find the name of each key, retrieve the value, and then normalize the name that will be used for output/environment variable.
If you want all secrets in JSON, you can currently use a separate action step to do that with the output as mentioned above.

**Comment by maxcoulombe** [comment]
Hello, I'm happy to report that this feature has been implemented in #488 . 
[Here is a link to the relevant documentation](https://github.com/hashicorp/vault-action?tab=readme-ov-file#multiple-secrets).

In short, it is not possible to fetch all keys from a given path with the wildcard character:
```
with:
    secrets: |
        secret/data/ci/aws * | MYAPP_ ;
```

Cheers!
